### DISCUSSION TITLE
loop through list of files

### DISCUSSION THREAD

**Discussion by ForestedDev** [community/community#25952] [Actions]
<p>how would i loop through a list of files and run a workflow on each file?<br>
say i have a list of files in yaml:</p>
<pre><code>&lt;font face="courier new,courier"&gt; files:
    scss: [fileone.scss,file2.sass]
    prettify: [codeone.js,code2.json]&lt;/font&gt;
</code></pre>
<p>on the scss files i must run one workflow that compiles them into css, and the one that prettifies them. and on the prettify files i just have to run the prettifier workflow. How would i loop through an array and pass it to a workflow so i can reference it with something like:</p>
<pre><code>${{ vars.FILE }}
</code></pre>

**Comment by airtower-luna** [comment]
<p>You could set them as <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/using-environment-variables" rel="nofollow noopener">environment variables</a> and use shell commands to loop over them, e.g. (Linux shell syntax, might be different on other platforms):</p>
<pre><code>steps:
  - name: run loop
    env:
      FILES: a.scss b.scss
    run: for f in $FILES; do [YOUR COMMAND HERE]; done
</code></pre>
<p>Though I think it might be better to use a build tool (e.g. make or whatever else suits your project) for this, or to add a script to your repository that does the processing based on parameters or even automatically discovered input.</p>

**Comment by weide-zhou** [comment]
<p>I’m not sure how you compile the scss files into css, but ‘<a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix" rel="nofollow noopener">jobs.&lt;job_id&gt;.strategy.matrix</a>’ could be an option for you.</p>
<p><img src="https://user-images.githubusercontent.com/52772761/181094287-80040f26-a71f-4dd8-b189-85f89a97efce.png" alt="matrixoption.png" data-base62-sha1="x0I2xj0RDiFNX1DTmWgpOdA5cWe" width="611" height="456"></p>

**Comment by ForestedDev** [comment]
<p>im using:</p>
<pre><code>- name: Sass Build
        uses: gha-utilities/sass-build@v0.1.6
</code></pre>
<p>to build it</p>

**Comment by ForestedDev** [comment]
<p>using this, how would i call a github action using a consle command?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/vortetty">@vortetty</a> ,</p>
<p>action ‘gha-utilities/sass-build’ has defined <a href="https://github.com/gha-utilities/sass-build/blob/2ec7f8ccdd56d897103280dfc135e50f306e42d9/action.yml#L5" rel="nofollow noopener">inputs</a> ‘source’ and ‘destination’, you can use matrix value for them.</p>
<p>Code sample as below:</p>
<pre><code>jobs:
  loopcheck:
    strategy:
      matrix:
        scssfiles: [fileone,file2]
    runs-on: [ubuntu-latest]
    steps:
      - name: test
        run: |
          echo ${{ matrix.scssfiles }}.scss
      - uses: actions/checkout@v1
      - name: Compile .css from .scss
        uses: gha-utilities/sass-build@v0.1.6
        with:
          source: scss/${{ matrix.scssfiles }}.scss
          destination: css/${{ matrix.scssfiles }}.css
</code></pre>
<p>It will complie the two scss files seperately into css folder. Hope it helps.</p>

**Comment by ForestedDev** [comment]
<p>Perfect, thank you!</p>

**Comment by rajanpanchal** [comment]
<p>what language it is for “for” loop?</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="airtower-luna" data-post="2" data-topic="17545">
<div class="title">
<div class="quote-controls"></div>
 airtower-luna:</div>
<blockquote>
<pre><code class="lang-auto">for f in $FILES; do [YOUR COMMAND HERE]; done
</code></pre>
</blockquote>
</aside>
<p>I guess you mean this? That’s sh (Unix/Linux shell). By default <code>run:</code> commands run in such a shell on Linux runners.</p>
